src/event_pusher/mod_event_pusher_rabbit.erl

```diff
@@ -85,6 +96,53 @@ publish_user_presence_change(JID, Status) ->
                                    exchange => PresenceExchange}},
                available_worker).
 
+-spec publish_user_chat_message_sent_event(From :: jid:jid(),
+                                           To :: jid:jid(),
+                                           Packet :: exml:element()) -> ok.
+publish_user_chat_message_sent_event(From = #jid{lserver = Host}, To, Packet) ->
+    Topic = opt(Host, chat_msg_sent_topic, ?DEFAULT_CHAT_MSG_SENT_TOPIC),
+    publish_user_chat_message_event(user_chat_msg_sent, #{from => From,
+                                                          to => To,
+                                                          packet => Packet,
+                                                          topic => Topic}).
+
+-spec publish_user_chat_message_received_event(From :: jid:jid(),
+                                               To :: jid:jid(),
+                                               Packet :: exml:element()) -> ok.
+publish_user_chat_message_received_event(From = #jid{lserver = Host}, To,
+                                         Packet) ->
+    Topic = opt(Host, chat_msg_recv_topic, ?DEFAULT_CHAT_MSG_RECV_TOPIC),
+    publish_user_chat_message_event(user_chat_msg_recv, #{from => From,
```

Comment:
I can see not point to use maps here. It is used just to pass parameters (All keys are mandatory). I think those options should be passed as separate arguments.

Reply:
I decided to use a map here because I'd like to avoid writing a function that accepts too many parameters. Without that it would be 5 arguments passed to the function. Apart from that I think that those arguments are connected with each other and make a logical whole. @bszaf do you see my point now?